### PR TITLE
Busybox CVE filter

### PR DESCRIPTION
**Pull request by lennarthenke**
This pull request introduces a new feature to filter BusyBox CVEs based on the components present in the binary file and the CVE summary specifying the vulnerability. However, the local CVE database only contains the summary for some entries, which is why requests are used to retrieve data from the NVD database. Unfortunately, this approach is not ideal, as sending too many requests at once results in a 403 error.
To address this issue, it is recommended to update the local database with the missing summaries, so that future code changes can rely on the local database instead of making requests to the NVD database.

### CODE REVIEW

File: src/plugins/analysis/cve_lookup/internal/busybox_cve_filter.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    if GROUP_1 is None or GROUP_2 is None:
+        group_1_path = Path(__file__).parent / 'group_1.txt'
+        group_2_path = Path(__file__).parent / 'group_2.txt'
+        with group_1_path.open('r') as f1, group_2_path.open('r') as f2:
+            GROUP_1 = f1.read().splitlines()
+            GROUP_2 = f2.read().splitlines()
+
+    pattern_1 = r'(?:\")(?:\-|\")'
+    pattern_2 = r'(?:\b|\(|\"|\_)(?:{})(?:\b|\)|-)'
```
[the remark this answers is not in the document]
Comment (lennarthenke) [reply src/plugins/analysis/cve_lookup/internal/busybox_cve_filter.py]: Thanks, the patterns are now pre-compiled.

File: src/plugins/analysis/cve_lookup/internal/busybox_cve_filter.py [hunk 2]
```diff
@@ -55,25 +32,30 @@ def get_busybox_components(file_object: FileObject) -> list[str]:
...
+            if any(word in components for word in matched_words):
+                filtered_cves.append(cve)
         else:
-            filtered_cves[cve_id] = ['General CVE']
+            filtered_cves.append(cve)
 
-    return remove_unrelated_cves(filtered_cves, components)
+    num_deleted = len(cves) - len(filtered_cves)
+    if num_deleted > 0:
+        logging.info(f'Deleted {num_deleted} CVEs with components not found in this BusyBox binary')
```
[the remark this answers is not in the document]
Comment (lennarthenke) [reply src/plugins/analysis/cve_lookup/internal/busybox_cve_filter.py]: I changed the logging to "debug" and include the file objects' UID now.

File: src/plugins/analysis/cve_lookup/internal/lookup.py [hunk 3]
```diff
@@ -16,14 +17,16 @@
     from .database.db_connection import DbConnection
     from .database.schema import Association, Cpe
     from collections.abc import Callable
+    from objects.file import FileObject
 
 VALID_VERSION_REGEX = re.compile(r'v?(\d+!)?\d+(\.\d+)*([.-]?(a(lpha)?|b(eta)?|c|dev|post|pre(view)?|r|rc)?\d+)?')
 
 
 class Lookup:
     DB_PATH = str(Path(__file__).parent / 'database/cve_cpe.db')
```
[the remark this answers is not in the document]
Comment (lennarthenke) [reply src/plugins/analysis/cve_lookup/internal/lookup.py]: Yes, I removed it.

File: src/plugins/analysis/cve_lookup/internal/busybox_cve_filter.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    return filter_cves_by_component(cves, components)
+
+
+def get_busybox_components(file_object: FileObject) -> list[str]:
+    """
+    Extracts the BusyBox components from the binary file.
+    """
+    file_path = Path(file_object.file_path)
+    with file_path.open(mode='rb') as f:
+        data = f.read()
```
[the remark this answers is not in the document]
Comment (lennarthenke) [reply src/plugins/analysis/cve_lookup/internal/busybox_cve_filter.py]: Thank you for the tip. I implemented the suggested change.

File: src/plugins/analysis/cve_lookup/test/data/busybox_1.32.0_x86_full
[the remark this answers is not in the document]
Comment (lennarthenke) [reply src/plugins/analysis/cve_lookup/test/data/busybox_1.32.0_x86_full:1]: I removed the busybox binaries and now only utilize their components for testing purposes.

**Review by jstucke — APPROVED** [review]
LGTM!

**Pull request merged**